students/ZackConnaughton/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,140 @@
+    with mongo:
+        database = mongo.connection.media
+
+        output_list = list(database.rentals.aggregate([
+            {'$match':
+                {'product_id': product_id}},
+            {'$lookup':
+                {'from': "customers",
+                'localField': "user_id",
+                'foreignField': "user_id",
+                'as': "cust"}},
+            {'$project':
+                {'_id': 0,
+                'user_id': 1,
+                'name': '$cust.name',
+                'address': '$cust.address',
+                'phone_number': '$cust.phone_number',
+                'email': '$cust.email'}}
+
+                ]))
+        return(output_list)
+
+
+def print_mdb_collection(collection_name):
+    for doc in collection_name.find({}):
+        print(doc)
+
+def csv_list(file):
+    return_list = []
+    with open('sample_csv_files/' + file, encoding='utf-8-sig') as csv_file:
+        file_input = csv.DictReader(csv_file)
+        for row in file_input:
+            return_list.append(row)
+    return return_list
+
+
+def drop_database_data(collection):
+    mongo = MongoDBConnection()
+
+    with mongo:
+        database = mongo.connection.media
+        database[collection].drop()
+        return True
+        #add something here if it cant be found
+
+def import_data(dir_name, product_csv, customer_csv, rental_csv):
+    mongo = MongoDBConnection()
+
+    with mongo:
+        database = mongo.connection.media
+
+        #Refactor at some point - reduce repetition
+        product_collection = database['products']
+        product_list = csv_list(product_csv)
+        product_list_count = len(product_list)
+        product_list_errors = 0
+        for item in product_list:
+            try:
+                product_collection.insert_one(item)
+            except:
+                #add log here for which record failed
+                product_list_errors += 1
+
+        customer_collection = database['customers']
+        customer_list = csv_list(customer_csv)
+        customer_list_count = len(customer_list)
+        customer_list_errors = 0
+        for item in customer_list:
+            try:
+                customer_collection.insert_one(item)
+            except:
+                customer_list_errors += 1
+
+        rental_collection = database['rentals']
+        rental_list = csv_list(rental_csv)
+        rental_list_count = len(rental_list)
+        rental_list_errors = 0
+        for item in rental_list:
+            try:
+                rental_collection.insert_one(item)
+            except:
+                rental_list_errors += 1
+
+        import_data_added = (product_list_count - product_list_errors,
+                             customer_list_count - customer_list_errors,
+                             rental_list_count - rental_list_errors)
+        import_data_errors = (product_list_errors,
+                              customer_list_errors,
+                              rental_list_errors)
+
```

Comment:
This function is pretty big, and it looks like you are doing basically the same thing several times. I would make a function so you don't have the same code written several times.